langfuse/openai.py

```diff
@@ -558,13 +558,13 @@ async def _wrap_async(
         openai_response = await wrapped(**arg_extractor.get_openai_args())
 
         if _is_streaming_response(openai_response):
-            return LangfuseResponseGeneratorAsync(
+            return aiter(LangfuseResponseGeneratorAsync(
```

Comment:
It raises an error if anext is called while the original openai implementation works properly.

Reply:
The class now implements __aiter__() but does not implement __anext__(), making it an iterable rather than an iterator.

To maintain compatibility with code that expects an iterator, I’ve added a call to aiter(langfuse) where the LangfuseResponseGeneratorAsync class is returned. This ensures that we’re explicitly returning an iterator, preserving the original behavior.